nsxt/provider.go

```diff
@@ -627,35 +628,9 @@ func configurePolicyConnectorData(d *schema.ResourceData, clients *nsxtClients)
 	}
 
 	if securityContextNeeded {
-		if len(vmcAccessToken) > 0 {
-			if vmcAuthHost == "" {
-				return fmt.Errorf("vmc auth host must be provided if auth token is provided")
-			}
-
-			apiToken, err := getAPIToken(vmcAuthHost, vmcAccessToken)
-			if err != nil {
-				return err
-			}
-
-			if vmcAuthMode == "Bearer" {
-				clients.CommonConfig.BearerToken = apiToken
-			} else {
-
-				securityCtx.SetProperty(security.AUTHENTICATION_SCHEME_ID, security.OAUTH_SCHEME_ID)
-				securityCtx.SetProperty(security.ACCESS_TOKEN, apiToken)
-			}
-		} else {
-			if username == "" {
-				return fmt.Errorf("username must be provided")
-			}
-
-			if password == "" {
-				return fmt.Errorf("password must be provided")
-			}
-
-			securityCtx.SetProperty(security.AUTHENTICATION_SCHEME_ID, security.USER_PASSWORD_SCHEME_ID)
-			securityCtx.SetProperty(security.USER_KEY, username)
-			securityCtx.SetProperty(security.PASSWORD_KEY, password)
+		err := configureSecurityContext(securityCtx, clients, vmcAccessToken, vmcAuthHost, vmcAuthMode, username, password)
```

Comment:
would it make sense to return `securityCtx` rather than passing it as a parameter?

Reply:
Also since VMC is not relevant for compute manager work, I'd suggest to leave the VMC part out of the function, which can be called `getBasicAuthSecurityContext` and become much simpler

Reply:
I think this is the provider's configurePolicyConnectorData function, and vmc parameters are still needed here?